app/helpers/visit_helper.rb

```diff
@@ -18,44 +16,45 @@ def prison_name
   end
 
   def prison_names
-    Rails.configuration.prison_data.keys.sort
+    Prison.names
   end
 
   def prison_data(source=visit)
-    Rails.configuration.prison_data.fetch(source.prisoner.prison_name.to_s)
+    prison_name = source.is_a?(String) ? source : source.prisoner.prison_name.to_s
+    Prison.find(prison_name)
   end
 
   def prison_phone
-    prison_data['phone']
+    prison_data.phone
   end
 
   def prison_email
-    prison_data['email']
+    prison_data.email
   end
 
   def prison_email_link
-    mail_to prison_data['email']
+    mail_to prison_data.email
   end
 
   def prison_postcode
-    prison_data['address'][-1]
+    prison_data.address[-1]
   end
 
   def prison_slot_anomalies
-    prison_data['slot_anomalies']
+    prison_data.slot_anomalies
   end
 
   def prison_address(glue='<br>'.html_safe)
-    safe_join(prison_data['address'], glue)
+    safe_join(prison_data.address, glue)
   end
 
   def adult_age
-    prison_data['adult_age'] || 18
+    prison_data.adult_age || 18
   end
 
   def prison_url(visit)
-    data = prison_data(visit)
-    slug = data['finder_slug'] ? data['finder_slug'] : visit.prisoner.prison_name.parameterize
+    return unless prison_data
```

Comment:
There's at least one spec (a view spec, I think) that breaks if this doesn't check for nil.   Might be better to just try(:finder_slug) instead.